### PR TITLE
[12.0][ADD] account_move_line_tax_single: only one tax per line

### PR DESCRIPTION
**Pull request by alexey-pelykh**
Limit to only 1 tax per AML

### CODE REVIEW

File: account_move_line_tax_single/hooks.py [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# Copyright 2020 CorporateHub (https://corporatehub.eu)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, SUPERUSER_ID
+
+
+def post_init_hook(cr, registry):
```
Comment (dreispt) [inline account_move_line_tax_single/hooks.py:7]: I don't think this is needed: when a computed field database column is created, is is automatically populated with computed values.

**Comment by marco-marchiori** [comment]
Why only 1 tax per line?
Where can I find the motivations for that?
What about v14? Is there a constraint like that in the community edition?

**Comment by alexey-pelykh** [comment]
@marco-marchiori Odoo by default (with is good) allows multiple taxes applied to a line, but in some usecases/businesses that's not really needed.

**Comment by marco-marchiori** [comment]
So 
- Odoo allows multiple taxes
- in some cases they are needed
- in other not.
So what's the rational of making a constraint?
I suppose that:
- if TAX stands for VAT the constraint could make sense
- if TAX stands for more (possible taxes) than VAT there could be a problem.

File: account_move_line_tax_single/tests/test_account_move_line.py [hunk 2]
```diff
@@ -0,0 +1,463 @@
...
+    #     self.assertEqual(len(statement.line_ids), 2)
+    #     fee_line = statement.line_ids - line
+
+    #     self.assertEqual(line.amount, -49.44)
+    #     self.assertEqual(line.amount_currency, -56.0)
+    #     self.assertEqual(line.currency_id, self.currency_usd)
+
+    #     self.assertEqual(fee_line.amount, -0.50)
+    #     self.assertEqual(fee_line.amount_currency, 0.0)
+    #     self.assertFalse(fee_line.currency_id)
```
Comment (dreispt) [inline account_move_line_tax_single/tests/test_account_move_line.py]: WIP?

**Comment by intero-chz** [comment]
Shouldn't there be something to prevent installation if there are invoice lines with more than one tax on it already?

File: account_move_line_tax_single/hooks.py (hunk 1 above)
Comment (alexey-pelykh) [reply account_move_line_tax_single/hooks.py:7]: Good point, thanks!

Comment (alexey-pelykh) [reply account_move_line_tax_single/hooks.py:7]: I'm just not sure how inverse will interact with compute in that case, like won't "None" value of new field won't get propagated first.

**Comment by alexey-pelykh** [comment]
@marco-marchiori yes, module adds constraint that's useful only in cases when business uses only VAT in their operations.
@intero-chz that actually can be done, and I guess should - to avoid install-by-mistake-and-loose-data

File: account_move_line_tax_single/tests/test_account_move_line.py (hunk 2 above)
Comment (alexey-pelykh) [reply account_move_line_tax_single/tests/test_account_move_line.py]: It was, not any more 👍

**Pull request closed without merging**